py/src/braintrust/logger.py

```diff
@@ -1288,16 +1296,16 @@ async def wrapper_async(*f_args, **f_kwargs):
                 return ret
 
         if inspect.iscoroutinefunction(f):
-            return wrapper_async
+            return cast(F, wrapper_async)
         else:
-            return wrapper_sync
+            return cast(F, wrapper_sync)
 
     # We determine if the decorator is invoked bare or with arguments by
     # checking if the first positional argument to the decorator is a callable.
     if len(span_args) == 1 and len(span_kwargs) == 0 and callable(span_args[0]):
-        return decorator(span_args[1:], span_kwargs, span_args[0])
+        return decorator(span_args[1:], span_kwargs, cast(span_args[0], F))
```

Comment:
isn't the argument order "type" then "value"?

Reply:
Yes! It failed a test internally as a result :)